src/VisualStudio/Core/Def/Implementation/ProjectSystem/AnalyzerConfigDocumentAsSolutionItemHandler.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using EnvDTE;
+using EnvDTE80;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Extensions;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.VisualStudio.Shell;
+using Roslyn.Utilities;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    [Export(typeof(AnalyzerConfigDocumentAsSolutionItemHandler)), Shared]
+    internal partial class AnalyzerConfigDocumentAsSolutionItemHandler : IDisposable
+    {
+        private static readonly string LocalRegistryPath = $@"Roslyn\Internal\{nameof(AnalyzerConfigDocumentAsSolutionItemHandler)}\";
+        private static readonly Option<bool> NeverShowAgain = new Option<bool>(nameof(AnalyzerConfigDocumentAsSolutionItemHandler), nameof(NeverShowAgain),
+            defaultValue: false, storageLocations: new LocalUserProfileStorageLocation(LocalRegistryPath + nameof(NeverShowAgain)));
+
+        private readonly VisualStudioWorkspace _workspace;
+        private readonly IThreadingContext _threadingContext;
+        private readonly DTE _dte;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public AnalyzerConfigDocumentAsSolutionItemHandler(
+            SVsServiceProvider serviceProvider,
+            VisualStudioWorkspace workspace,
+            IThreadingContext threadingContext)
+        {
+            _workspace = workspace;
+            _threadingContext = threadingContext;
+            _dte = (DTE)serviceProvider.GetService(typeof(DTE));
```

Comment:
❗️ MEF constructors must not have threading dependencies (including ones from COM)

Reply:
Thanks, fixed.